frontend/login.html

```diff
@@ -0,0 +1,30 @@
+<!DOCTYPE html>
+<html>
+    <head>
+        <meta charset="utf-8">
+        <meta http-equiv="X-UA-Compatible" content="IE=edge">
+        <title>marsRover</title>
+        <meta name="description" content="">
+        <meta name="viewport" content="width=device-width, initial-scale=1">
+        <link rel="stylesheet" href="styles.css">
+    </head>
+    <body>
+        <h1>Mars Rover Project</h1>
+        <h2>Login</h2>
+        <p>Login here to access your saved NASA images and data.</p>
+        <div class="login">
+            <div class="form-input">
+                <label for="username">Username</label>
+                <input type="text" id="username">
+            </div>
+            <div class="form-input">
+                <label for="password">Password</label>
+                <input type="password" id="password">
+            </div>
+            <button type="submit">
+                Submit
+            </button>
+        </div>
+        <script src="" async defer></script>
```

Comment:
Ah, I just noticed your use of async and defer. Good stuff, what do they do and what's the difference?

Reply:
:) To be perfectly honest, I used a boiler plate for this html.
Looking them up, it seems that they improve page loading efficiency with async used mainly in modern browsers and defer used as a fallback for older browsers if both are used in the script tag.
They seem to be most effective when the script tag is in the head - so I'm not sure why the boiler plate includes them here. 
From reading around this, they work by reducing the load time of the page:
* Async fetches the script while the head HTML is parsed so the only wait time is when the script is executed - the HTML body is parsed after this happens.
* Defer also fetches the script asynchronously but delays the execution of the script until the entirety of the HTML is parsed, so no pause in loading would be observed. I'm not sure how this would differ from just placing the script tag at the bottom of the body?